flexmeasures/ui/crud/api_wrapper.py

```diff
@@ -41,17 +41,28 @@ def _maybe_raise(
         if response.status_code not in do_not_raise_for:
             response.raise_for_status()
 
+    def _url_root(self) -> str:
+        """
+        Get the root for the URLs this API should use to call FlexMeasures.
+        """
+        url_root = request.url_root
+        if current_app.config.get("FLEXMEASURES_FORCE_HTTPS", False):
+            # this replacement is for the case we are behind a load balancer who talks http internally
+            url_root = url_root.replace("http://", "https://")
+        return url_root
+
     def get(
         self,
         url: str,
         query: dict[str, Any] | None = None,
         do_not_raise_for: list | None = None,
     ) -> requests.Response:
+        url_root = self._url_root()
```

Comment:
Maybe, you can already build the final URL here and use it in the debug and requests.get. Similar in the other methods.